features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/AlarmSearchProvider.java

```diff
@@ -86,8 +85,16 @@ public String getSearchProviderNamespace() {
     public boolean contributesTo(String namespace) {
         return "nodes".equals(namespace);
     }
-    
-    public class AlarmSearchResult extends SearchResult {
+
+	@Override
+	public Criteria getCriteriaFromQuery(SearchResult input) {
+		// TODO RS Verify that this actually works... node label seems to be missing
+		AlarmHopCriteria criteria = createCriteria(input);
```

Comment:
the `createCriteria` method should set the id of the criteria based on the input's id. Then the next line can be removed.